[DOCS] Add system & hidden indices definitions. (#64834)
elastic/elasticsearch: docs/reference/glossary.asciidoc

Searching a frozen index is lower overhead than re-opening a closed index to enable searching.
// end::frozen-index-def[]

[[glossary-hidden-index]] hidden index ::
// tag::hidden-index-def[]
An index that is excluded by default when you access indices using a wildcard expression. 

> To include hidden indices, you must specify the `expand_wildcards` parameter.

Reviewer: One minor point that it might be good to mention here: Hidden indices are also included for wildcard expressions that start with a dot, e.g. `.watcher-history*`. This was done in part to mirror the Unix globbing behavior for dot files.

> An index that is excluded by default when you access indices using a wildcard expression. 
> To include hidden indices, you must specify the `expand_wildcards` parameter.

Author: ```suggestion
An index that is excluded by default when you access indices using a wildcard expression. 
To include hidden indices, you must specify the `expand_wildcards` parameter.
```
```suggestion
An index that is excluded by default when you access indices using a wildcard expression. 
To include hidden indices, you must specify the `expand_wildcards` parameter.
```
```suggestion
An index that is excluded by default when you access indices using a wildcard expression. 
You can specify the `expand_wildcards` parameter to include hidden indices.
Note that hidden indices _are_ included if the wildcard expression starts with a dot, for example `.watcher-history*`.
```

See the {ref}/indices-split-index.html[split index API].
// end::split-def[]

[[glossary-system-index]] system index ::

> // tag::hidden-index-def[]

Reviewer 2: ```suggestion
// tag::system-index-def[]
```

An index that contains configuration information or other data used internally by the system, 
such as the `.security` index

> You cannot directly access or modify system indices.

Reviewer: I'm not sure if this is relevant because I'm not sure what branches this is aimed at, but this will only be true for 8.0+, for 7.x you can access/modify system indices, but doing so is deprecated.

> You cannot directly access or modify system indices.

Author: ```suggestion
You should not directly access or modify system indices.
```

See the {ref}/indices-split-index.html[split index API].
// end::split-def[]

[[glossary-system-index]] system index ::
// tag::system-index-def[]

> An index that contains configuration information or other data used internally by the system, 
> such as the `.security` index
> You should not directly access or modify system indices.

Author: ```suggestion
An index that contains configuration information or other data used internally by the system, 
such as the `.security` index. 
The name of a system index is always prefixed with a dot.
You should not directly access or modify system indices.
```
